kpi_dashboard/models/kpi_kpi.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2020 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+import ast
+import random
+
+
+class KpiKpi(models.Model):
+    _name = "kpi.kpi"
+    _description = "Kpi Kpi"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    cron_id = fields.Many2one("ir.cron", readonly=True, copy=False)
+    computation_method = fields.Selection(
+        [("function", "Function")], required=True
+    )
+    value = fields.Serialized()
+    dashboard_ids = fields.One2many("kpi.dashboard.kpi", inverse_name="kpi_id")
+    model_id = fields.Many2one("ir.model",)
+    function = fields.Char()
+    args = fields.Char()
+    kwargs = fields.Char()
+    widget = fields.Selection(
+        [("number", "Number"), ("meter", "Meter"), ("graph", "Graph")],
+        required=True,
+        default="number",
+    )
+    value_last_update = fields.Datetime(readonly=True)
+    prefix = fields.Char()
+    suffix = fields.Char()
+    action_id = fields.Many2one("ir.actions.actions")
+
+    def _cron_vals(self):
+        return {
+            "name": self.name,
+            "model_id": self.env.ref("kpi_dashboard.model_kpi_kpi").id,
+            "interval_number": 1,
+            "interval_type": "hours",
+            "state": "code",
+            "code": "model.browse(%s).compute()" % self.id,
+            "active": True,
+        }
+
+    def compute(self):
+        for record in self:
+            record._compute()
+        return True
+
+    def _compute(self):
+        self.write(
+            {
+                "value": getattr(
+                    self, "_compute_value_%s" % self.computation_method
+                )()
+            }
+        )
+        notifications = []
+        for dashboard_kpi in self.dashboard_ids:
+            channel = "kpi_dashboard_%s" % dashboard_kpi.dashboard_id.id
+            notifications.append([channel, dashboard_kpi._read_dashboard()])
+        if notifications:
+            self.env["bus.bus"].sendmany(notifications)
+
+    def _compute_value_function(self):
+        obj = self
+        if self.model_id:
+            obj = self.env[self.model_id.model]
+        args = ast.literal_eval(self.args or "[]")
+        kwargs = ast.literal_eval(self.kwargs or "{}")
+        return getattr(obj, self.function)(*args, **kwargs)
+
+    def test_demo_number(self):
+        return {
+            "value": random.random() * 10000,
+            "previous": random.random() * 10000,
+        }
+
+    def test_demo_meter(self):
```

Comment:
I think demo data should not be in model files.

Reply:
If you don't have any other work around, you can create an extra module kpi_dashboard_test for that contains that part of code.

Reply:
It is better to add a second module. True.